editor.planx.uk/src/@planx/components/PlanningConstraints/Public.tsx

```diff
@@ -71,79 +74,80 @@ function Component(props: Props) {
+
   // Get the WKT representation of the site boundary drawing or address point to pass to Planning Data
   const wktPoint = `POINT(${longitude} ${latitude})`;
   const wktPolygon: string | undefined =
     siteBoundary && stringify(siteBoundary);
-
   const planningDataParams: Record<string, string> = {
     geom: wktPolygon || wktPoint,
+    vals: dataValues.join(","),
     ...params,
   };
 
-  // Fetch planning constraints data for a given local authority
-  const root = `${import.meta.env.VITE_APP_API_URL}/gis/${teamSlug}?`;
+  // Fetch planning constraints data for a given local authority if Planning Data & a geometry is available
+  const shouldFetchPlanningData =
+    hasPlanningData &&
+    latitude &&
+    longitude &&
+    dataValues.some((val) => val !== "road.classified");
   const teamGisEndpoint: string =
-    root +
-    new URLSearchParams(
-      hasPlanningData ? planningDataParams : undefined,
-    ).toString();
-
-  const fetcher: Fetcher<GISResponse | GISResponse["constraints"]> = (
-    url: string,
-  ) => fetch(url).then((r) => r.json());
+    `${import.meta.env.VITE_APP_API_URL}/gis/${teamSlug}?` +
+    new URLSearchParams(planningDataParams).toString();
   const {
     data,
     mutate,
     error: dataError,
     isValidating,
-  } = useSWR(() => (latitude && longitude ? teamGisEndpoint : null), fetcher, {
-    revalidateOnFocus: false,
-  });
+  } = useSWR(
+    () => (shouldFetchPlanningData ? teamGisEndpoint : null),
+    fetcher,
+    { revalidateOnFocus: false },
+  );
 
   // If an OS address was selected, additionally fetch classified roads (available nationally) using the USRN identifier,
   //   skip if the applicant plotted a new non-UPRN address on the map
+  const shouldFetchRoads =
+    hasPlanningData && usrn && dataValues.includes("road.classified");
   const classifiedRoadsEndpoint: string =
     `${import.meta.env.VITE_APP_API_URL}/roads?` +
     new URLSearchParams(usrn ? { usrn } : undefined)?.toString();
-
   const {
     data: roads,
     error: roadsError,
     isValidating: isValidatingRoads,
   } = useSWR(
-    () => (usrn && hasPlanningData ? classifiedRoadsEndpoint : null),
+    () => (shouldFetchRoads ? classifiedRoadsEndpoint : null),
     fetcher,
     { revalidateOnFocus: false },
   );
 
   // Merge Planning Data and Roads responses for a unified list of constraints
-  const constraints: GISResponse["constraints"] | Record<string, any> = {
+  const constraints: GISResponse["constraints"] = {
     ...data?.constraints,
     ...roads?.constraints,
   };
-
-  const metadata: GISResponse["metadata"] | Record<string, any> = {
+  const metadata: GISResponse["metadata"] = {
     ...data?.metadata,
     ...roads?.metadata,
   };
 
   const handleSubmit = () => {
     // `_constraints` & `_overrides` are responsible for auditing
-    const _constraints: Array<
-      EnhancedGISResponse | GISResponse["constraints"]
-    > = [];
+    const _constraints: Array<EnhancedGISResponse> = [];
```

Comment:
There's a couple follow-on type tidy ups here that I missed when retiring custom GIS in #3977